### PR TITLE
Allow support for JSON arrays within annotations

### PR DESCRIPTION
**Pull request by rlweb**
As said, added square brackets to complete support for JSON.

### CODE REVIEW

File: tests/Doctrine/Tests/Common/Annotations/DocParserTest.php [hunk 1]
```diff
@@ -422,6 +422,8 @@ public function getAnnotationVarTypeProviderValid()
               // array type
             array('array', '{@AnnotationExtendsAnnotationTargetAll}'),
             array('array', '{@AnnotationExtendsAnnotationTargetAll,@AnnotationExtendsAnnotationTargetAll}'),
+            array('array', '[@AnnotationExtendsAnnotationTargetAll]'),
+            array('array', '[@AnnotationExtendsAnnotationTargetAll,@AnnotationExtendsAnnotationTargetAll]'),
```
Comment (Ocramius) [inline tests/Doctrine/Tests/Common/Annotations/DocParserTest.php]: Would need a few tests around parsing array elements, their order, and different types of sub-elements

Comment (Ocramius) [inline tests/Doctrine/Tests/Common/Annotations/DocParserTest.php]: Also nested arrays

File: lib/Doctrine/Common/Annotations/DocLexer.php [hunk 2]
```diff
@@ -51,6 +51,8 @@
     const T_TRUE                = 110;
     const T_NULL                = 111;
     const T_COLON               = 112;
+    const T_OPEN_SQR_BRACKETS   = 113;
```
Comment (Ocramius) [inline lib/Doctrine/Common/Annotations/DocLexer.php]: `SQUARE`

File: lib/Doctrine/Common/Annotations/DocLexer.php [hunk 3]
```diff
@@ -51,6 +51,8 @@
...
+    const T_CLOSE_SQR_BRACKETS  = 114;
```
Comment (Ocramius) [inline lib/Doctrine/Common/Annotations/DocLexer.php]: `SQUARE`

**Review by Ocramius — CHANGES_REQUESTED** [review]
Code looks good, but we need to improve the testing of nested values, optional additional values.

This should be done both in the parser tests, and in the annotation reader afterwards, in my opinion.

@guilhermeblanco any opinion on supporting this?

**Comment by Majkl578** [comment]
This is something that should be relatively easy with the new parser in Annotations 2.0. Milestoning, since there is no relevant issue, but it will require different implementation.

**Pull request closed without merging**